### PR TITLE
Update install and invite, add patreon and disclaimer.

### PR DESCRIPTION
**Pull request by scragly**
## **Please check if the PR fulfills these requirements**
- [ ] Tests have been conducted on the PR
- [x] Docs have been added / updated

## **What kind of change does this PR introduce?**

Updated the documentation site to use the new installation instruction (using PyPI), updates the support server URL, adds the twitch logo disclaimer, and adds the patreon support link.

### CODE REVIEW

File: docs/twitchio.rst [hunk 1]
```diff
@@ -4,26 +4,25 @@ TwitchIO
 Support
 ---------------------------
 For support using TwitchIO, please join the official `support server
-<http://discord.gg/JhW28zp>`_ on `Discord <https://discordapp.com/>`_.
+<https://discord.gg/RAKc3HF>`_ on `Discord <https://discordapp.com/>`_.
+
+`Become a patreon <https://twitchio.readthedocs.io/en/rewrite/twitchio.html>`_ and help support TwitchIO's development <3.
```
Comment (SnowyLuma) [inline docs/twitchio.rst]:
```suggestion
`Become a patreon <https://www.patreon.com/twitchio>`_ and help support TwitchIO's development <3.
```

Comment (scragly) [reply docs/twitchio.rst]: That's weird, I swear I copied that one. Thanks for fixing it :)

Comment (SnowyLuma) [reply docs/twitchio.rst]: It should also be `patron` instead of `patreon`, seems my edit didn't make it.

Comment (scragly) [reply docs/twitchio.rst]: Sorted and added coauthor manually.

**Pull request closed without merging**